core/validatorapi/validatorapi.go

```diff
@@ -58,7 +58,7 @@ func NewComponentInsecure(_ *testing.T, eth2Cl eth2wrap.Client, shareIdx int) (*
 //
 //nolint:gocognit
 func NewComponent(eth2Cl eth2wrap.Client, allPubSharesByKey map[core.PubKey]map[int]*bls_sig.PublicKey,
-	shareIdx int, feeRecipientAddress string, builderAPI bool, seenPubkeys func(core.PubKey),
+	shareIdx int, feeRecipientAddressByPubkey map[core.PubKey]string, builderAPI bool, seenPubkeys func(core.PubKey),
```

Comment:
same here, rather not pass in pubkeys twice, wouldn't it be better to use a single map with both shares and feerecipients? The other option is pass in a callback for `feeRecipientFunc func(core.Pubkey)string`

Reply:
Thanks for pointing out, went with `feeRecipientFunc func(core.Pubkey)string`